bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/scheduler/EventForegroundTask.kt

```diff
@@ -0,0 +1,50 @@
+package io.bucketeer.sdk.android.internal.scheduler
+
+import io.bucketeer.sdk.android.internal.di.Component
+import io.bucketeer.sdk.android.internal.event.EventInteractor
+import io.bucketeer.sdk.android.internal.event.SendEventsResult
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+
+internal class EventForegroundTask(
+  private val component: Component,
+  private val executor: ScheduledExecutorService,
+) : ScheduledTask {
+
+  override var isStarted: Boolean = false
+    private set
+
+  private var scheduledFuture: ScheduledFuture<*>? = null
+
+  private val eventUpdateListener = EventInteractor.EventUpdateListener { _ ->
+    executor.execute {
+      val result = component.eventInteractor.sendEvents(force = false)
+      if (result is SendEventsResult.Success && result.sent) {
+        reschedule()
+      }
+    }
+  }
+
+  internal fun reschedule() {
+    scheduledFuture?.cancel(false)
+    scheduledFuture = executor.scheduleWithFixedDelay(
```

Comment:
Just confirming, when it starts the scheduler, it calls once `component.eventInteractor.sendEvents(force = true)` in the beginning? Or only after the interval has reached?

Reply:
`ScheduledTask`s are managed in `TaskScheduler`

https://github.com/ca-dp/bucketeer-android-sdk/pull/71/files#diff-9fcb60929f5bfa8f596d022bc300e4d96df8a62e077b930a14b18e41dc055236R19

and `TaskScheduler`(which implements `DefaultLifecycleObserver`) is registered to `ProcessLifecycleOwner` in `BKTClientImpl#initializeInternal()`

https://github.com/ca-dp/bucketeer-android-sdk/pull/71/files#diff-933843224cca039e7d3bbd7550c39d420051db3ac75ba15284cb3adca143ff45R119

Reply:
Sorry, I meant the code to be executed in the `executor.scheduleWithFixedDelay` is called in the beginning once, or is it only called after the interval has been reached?

Reply:
Oh I see. It will be called after the interval.

second parameter of `cheduleWithFixedDelay()` is initialDelay, so we can call immediately if you want by setting it to `0` or some smaller value.

Reply:
Ah ok, there is an option to configure in case we want to call it immediately. Thanks.
I asked because if the `fetchEvaluationsSync` fails when the app is initialized, it will take 10 minutes until it updates the client cache.

Sometimes when we have a spike on the server due to a push notification on the client, too many requests come simultaneously, and the server autoscaling may not scale in time. In that case, some requests may fail, and it will take 10 minutes until it requests again.

That's why I asked about the retry process in the other comment.